components/Collections.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  Button,
+  ButtonGroup,
+  Spinner,
+  Stack,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import { getServicesList } from 'api'
+import { ServicesList } from 'models'
+import { UserDoc } from 'models/users'
+import React, { useEffect, useState } from 'react'
+import CollectionItem from './CollectionItem'
+
+interface CollectionProps {
+  userData: UserDoc
+}
+
+export const Collections = ({ userData }: CollectionProps): JSX.Element => {
+  const [loading, setLoading] = useState(true)
+  const [listData, setListData] = useState<ServicesList[]>([])
+
+  useEffect(() => {
+    const promises = []
+    for (const list of userData.lists) {
+      const listData = getServicesList(list)
+      promises.push(listData)
+    }
+
+    Promise.all(promises.map((p) => p.catch((e) => e))).then((list) => {
+      setListData(list.filter((l) => !(l instanceof Error)))
+      setLoading(false)
+    })
+  }, [userData.lists])
+
+  if (loading) {
+    return (
+      <Stack w="100%" alignItems={'center'}>
+        <Spinner variant="primary" />
+      </Stack>
+    )
+  }
```

Comment:
Thanks for finally adding some loading UI! Hopefully we can get designs for loading states at some point but this should work in the meantime.

Reply:
💯